src/main/java/org/spongepowered/api/item/inventory/InventoryTransformation.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.item.inventory.query.QueryOperation;
+import org.spongepowered.api.util.ResettableBuilder;
+
+/**
+ * A Inventory Transformation can filter and order an inventory
+ * <p>
+ *     See {@link InventoryTransformations} for some predefined Transformations
+ * </p>
+ */
+public interface InventoryTransformation {
+
+    /**
+     * Transforms an Inventory
```

Comment:
Please add a little more description on how this inventory is supposed to be mutated. How does it behave in relation to duplicate slots and stuff alike. (and if there are duplicate slots, how the user can omit them).

Either here or on class level.

Reply:
It really depends on the implementation of the specific transformation.
But I explained the ones you can create yourself with the builder.